src/mito2/src/read/scan_region.rs

```diff
@@ -220,11 +227,9 @@ impl ScanRegion {
     }
 
     /// Unordered scan.
-    pub(crate) fn unordered_scan(self) -> Result<UnorderedScan> {
+    pub(crate) async fn unordered_scan(self) -> Result<UnorderedScan> {
         let input = self.scan_input(true)?;
-        let scan = UnorderedScan::new(input);
-
-        Ok(scan)
+        UnorderedScan::new(input).await
```

Comment:
Can you postpond reading parquet meta to `UnorderedScan::build_stream` like how `SeqScan` does? To avoid I/O in the plan phase

Reply:
It's hard to distribute row groups to partitions without reading the metadata. Since we might read different numbers of row groups from different files. I'll also modify the `SeqScan` in the next PR......

We will cache the metadata after we read it so we won't do I/O next time.

But I'll find some ways to address this later.